tiup: update docs of tiup-bench (#11430)
pingcap/docs: tiup/tiup-bench.md

> 

Reviewer: L3 的 summary 也改下：
Learn how to stress test TiDB with TPC-C, TPC-H, CH, RawSQL, and YCSB workloads using TiUP

# Stress Test TiDB Using TiUP Bench Component

When you test the performance of a database, it is often required to stress test the database. To facilitate this, TiUP has integrated the bench component, which provides multiple workloads for stress testing. You can access these workloads by the following commands:

```bash
tiup bench tpcc   # Benchmark a database using TPC-C
tiup bench tpch   # Benchmark a database using TPC-H
tiup bench ch     # Benchmark a database using CH-benCHmark
tiup bench ycsb   # Benchmark a database using YCSB
tiup bench rawsql # Benchmark a database using arbitrary SQL files
```

> `tpcc`, `tpch`, `ch`, `rawsql` share the following common command flags. However, `ycsb` is mainly configured by a `.properties` file, which is described in its [usage guide](https://github.com/pingcap/go-ycsb#usage).

Reviewer: ```suggestion
`tpcc`, `tpch`, `ch`, and `rawsql` share the following common command flags. However, `ycsb` is mainly configured by a `.properties` file, which is described in its [usage guide](https://github.com/pingcap/go-ycsb#usage).
```

```
  -t, --acThreads int         OLAP client concurrency, only for CH-benCHmark (default 1)
      --conn-params string    Session variables, e.g. --conn-params tidb_isolation_read_engines='tiflash' for TiDB, --conn-params sslmode=disable for PostgreSQL
      --count int             Total execution count, 0 means infinite
  -D, --db string             Database name (default "test")
  -d, --driver string         Database driver: mysql, postgres (default "mysql")

>       --dropdata              Cleanup data before prepare

Reviewer 2: ```suggestion
      --dropdata              Clean up historical data before preparing 
```

  -H, --host strings          Database host (default to [127.0.0.1])

>       --ignore-error          Ignore error when running workload

Reviewer: ```suggestion
      --ignore-error          Ignore errors when running workload
```

      --interval duration     Output interval time (default 10s)
      --isolation int         Isolation Level 0: Default, 1: ReadUncommitted,

>                               2: ReadCommitted, 3: WriteCommitted, 4: RepeatableRead,
>                               5: Snapshot, 6: Serializable, 7: Linerizable

Reviewer 2: ```suggestion
                              2: ReadCommitted; 3: WriteCommitted; 4: RepeatableRead; 
                              5: Snapshot; 6: Serializable; 7: Linerizable)
```

      --max-procs int         runtime.GOMAXPROCS of golang, the limits of how many cores can be used

>       --output string         Output style, valid values can be { plain | table | json } (default "plain")

Reviewer 2: ```suggestion
      --output string         Output style. Valid values can be { plain | table | json } (default "plain")
```

  -p, --password string       Database password
  -P, --port ints             Database port (default to [4000])
      --pprof string          Address of pprof endpoint

>       --silence               Don't print error when running workload

Reviewer: ```suggestion
      --silence               Don't print errors when running workload
```

  -S, --statusPort int        Database status port (default to 10080)
  -T, --threads int           Thread concurrency (default to 1)
      --time duration         Total execution time (default to 2562047h47m16.854775807s)
  -U, --user string           Database user (default to "root")
```

> - You can pass comma-separated values to `--host` and `--port` to enable client-side load balancing. For example, when specifying `--host 172.16.4.1,172.16.4.2 --port 4000,4001`, the program will connect to 172.16.4.1:4000, 172.16.4.1:4001, 172.16.4.2:4000, 172.16.4.2:4001, chosen in round-robin fashion.

Reviewer: ```suggestion
- You can pass comma-separated values to `--host` and `--port` to enable client-side load balancing. For example, when you specify `--host 172.16.4.1,172.16.4.2 --port 4000,4001`, the program will connect to 172.16.4.1:4000, 172.16.4.1:4001, 172.16.4.2:4000, and 172.16.4.2:4001, chosen in round-robin fashion.
```

- `--conn-params` must follow the format of [query string](https://en.wikipedia.org/wiki/Query_string). Different databases might have different parameters. For example:
    - `--conn-params tidb_isolation_read_engines='tiflash'` forces TiDB to read from TiFlash.
    - `--conn-params sslmode=disable` disables SSL when you connect to PostgreSQL.

> - When running CH-benCHmark, you can use `--ap-host`, `--ap-port`, `--ap-conn-params` to specify a standalone TiDB server for OLAP queries.

Reviewer: ```suggestion
- When running CH-benCHmark, you can use `--ap-host`, `--ap-port`, and `--ap-conn-params` to specify a standalone TiDB server for OLAP queries.
```

  run         Run workload

Flags:
      --check-all            Run all consistency checks

>   -h, --help                 help for tpcc

Reviewer 2: ```suggestion
  -h, --help                 Help for TPC-C
```

>       --partition-type int   Partition type: 1 - HASH, 2 - RANGE, 3 - LIST (HASH-like), 4 - LIST (RANGE-like) (default 1)

Reviewer 2: ```suggestion
      --partition-type int   Partition type: 1 - HASH, 2 - RANGE, 3 - LIST (HASH-like), 4 - LIST (RANGE-like) (default to 1)
```

>       --parts int            Number to partition warehouses (default 1)

Reviewer 2: 请教下，这个参数是代表“各个分区里仓库的数量”吗，即 Number of warehouses in each partition？

Author: 不是，是 Number of partitions 的意思...

>       --warehouses int       Number of warehouses (default 10)

Reviewer 2: ```suggestion
      --warehouses int       Number of warehouses (default to 10)
```

```

### Test procedures

> Here we show simplified steps for running a TPC-C test. Please refer to [How to Run TPC-C Test on TiDB](/benchmark/benchmark-tidb-using-tpcc.md) for a more complete guide.

Reviewer 2: ```suggestion
The following provides simplified steps for running a TPC-C test. For detailed steps, see [How to Run TPC-C Test on TiDB](/benchmark/benchmark-tidb-using-tpcc.md).
```

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 --time 10m run
    ```

3. Check the consistency:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 check
    ```

4. Clean up data:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 cleanup
    ```

> Preparing data via SQL might be slow when you want to run a benchmark with a large data set. In that case, you can generate data in the CSV format by the following commands and then import it to TiDB via [TiDB lightning](/tidb-lightning/tidb-lightning-overview.md).

Reviewer: ```suggestion
Preparing data via SQL might be slow when you want to run a benchmark with a large data set. In that case, you can generate data in the CSV format by the following commands and then import it to TiDB via [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md).
```
